app/models/patron_request.rb

```diff
@@ -46,16 +46,63 @@ def default_service_point_code
     service_points.first || Settings.folio.default_service_point
   end
 
-  private
+  def mediateable?
+    items_in_location.all?(&:mediateable?)
+  end
+
+  def requires_needed_date?
+    return false if origin_location_code == 'PAGE-MP' || origin_location_code == 'SAL3-PAGE-MP'
+
+    true
+  end
+
+  def use_in_library?
+    items_in_location.all? { |item| item.effective_location.discovery_display_name.include?('For use in') }
+  end
+
+  def single_location_display
+    Folio::Types.service_points.find_by(code: pickup_destinations.first)
+  end
+
+  def single_location_label
+    return 'Must be used in library' if mediateable? || use_in_library?
+
+    'Will be delivered to'
+  end
+
+  def destination_library_code
+    find_location_library(single_location_display.code) if single_location_display
+  end
+
+  def find_location_library(lib_code)
+    Folio::Types.libraries.find_by(code: lib_code)&.code
+  end
+
+  def earliest_delivery_estimate
+    paging_info = PagingSchedule.for(self)
+    { 'date' => Time.zone.today + paging_info.days_later, 'time' => paging_info.will_arrive_text }
```

Comment:
I don't think this is right, because it doesn't take into account the library's open/closed days. The `PagingScheduleController` uses `earliest_delivery_estimate` which has that logic.

Reply:
You are correct. I updated     paging_info = PagingSchedule.for(self) to paging_info = PagingSchedule.for(self).earliest_delivery_estimate and now it should be right?